src/main/msp/msp.c

```diff
@@ -1344,12 +1344,17 @@ static bool mspProcessOutCommand(uint8_t cmdMSP, sbuf_t *dst)
         sbufWriteU8(dst, 1); // Rate numerator, not used anymore
         sbufWriteU8(dst, blackboxGetRateDenom());
         sbufWriteU16(dst, blackboxConfig()->p_ratio);
+        //Added in MSP API 1.42
+        sbufWriteU8(dst, systemConfig()->debug_mode);
```

Comment:
Looking at this in more detail, I think `MSP_BLACKBOX_CONFIG`, and having this conditional to `USE_BLACKBOX` isn't the right place for this - the `debug[]` facility is not only used for blackbox, it can also be used for live display in configurator.
I propose for this to be moved to `MSP_ADVANCED_CONFIG` / `MSP_SET_ADVANCED_CONFIG`, but open for other suggestions.

Reply:
Good point, makes much more sense to have it in `MSP_ADVANCED_CONFIG`

Reply:
Looking at this now, `MSP_ADVANCED_CONFIG` has 7 things that are still not implemented in the configurator. All from `motorConfig()->dev.motorPwmInversion` to `gyroConfig()->checkOverflow` are not in use currently, so adding debug modes at the end requires me to add all these above to the configurator. Would that be a good option or should I just put debug modes just above these that are not in use?

Reply:
No, append them at the end - we don't know if there are other MSP clients than configurator using these settings.
The configurator side implementation is not too involved - store the values in `PID_ADVANCED_CONFIG` on `MSP_ADVANCED_CONFIG`, and read them back from there in `MSP_SET_ADVANCED_CONFIG`.